tee-worker/ts-tests/integration-tests/dr_vc.test.ts

```diff
@@ -147,24 +148,39 @@ describe('Test Vc (direct request)', function () {
             );
             const eventsPromise = subscribeToEventsWithExtHash(requestIdentifier, context);
 
-            const requestVcCall = await createSignedTrustedCallRequestVc(
-                context.api,
-                context.mrEnclave,
-                context.api.createType('Index', nonce),
-                new PolkadotSigner(context.substrateWallet.alice),
-                aliceSubstrateIdentity,
-                context.api.createType('Assertion', assertion).toHex(),
-                context.api.createType('Option<RequestAesKey>', aesKey).toHex(),
-                requestIdentifier
-            );
+            let requestVcCall;
+            if (Array.isArray(assertion)) {
+                requestVcCall = await createSignedTrustedCallRequestBatchVc(
+                    context.api,
+                    context.mrEnclave,
+                    context.api.createType('Index', nonce),
+                    new PolkadotSigner(context.substrateWallet.alice),
+                    aliceSubstrateIdentity,
+                    context.api.createType('Vec<Assertion>', assertion).toHex(),
+                    context.api.createType('Option<RequestAesKey>', aesKey).toHex(),
+                    requestIdentifier
+                );
+            } else {
+                requestVcCall = await createSignedTrustedCallRequestVc(
+                    context.api,
+                    context.mrEnclave,
+                    context.api.createType('Index', nonce),
+                    new PolkadotSigner(context.substrateWallet.alice),
+                    aliceSubstrateIdentity,
+                    context.api.createType('Assertion', assertion).toHex(),
+                    context.api.createType('Option<RequestAesKey>', aesKey).toHex(),
+                    requestIdentifier
+                );
+            }
 
             const isVcDirect = true;
-            // const res = await sendRequestFromTrustedCall(context, teeShieldingKey, requestVcCall, isVcDirect);
-
             // Instead of waiting for final response we will listen all responses from the call
             const onMessageReceived = async (res: WorkerRpcReturnValue) => {
                 // if response is a A1 or A2, etc....
-                await assertVc(context, aliceSubstrateIdentity, res.value);
+                const vcresponse = context.api.createType('VCResponse', res.value);
```

Comment:
I'm a bit iffy on the `VCResponse` naming here:

`WorkerRpcReturnValue` contains `VCResponse` which contains either `RequestVCResult` if it's `len` prop is > 0, or String with the error msg, otherwise.

How about something more idiomatic like `RequestVcResultOrError`.

Also do I understand correctly that adding the `idx` into `WorkerRpcReturnValue` is not possible and thus we had to introduce a new layer called `VCResponse`?

Reply:
Hi @jonalvarezz `idx` is also include within this VCResponse. Same layer as `len`. Is it OK?

Reply:
Yes. I can renaming as you suggested.

Reply:
> idx is also include within this VCResponse. Same layer as len. Is it OK?

That was my second question. VCResult didn't exist before so we are purposely adding it. Do we need to because adding idx to `WorkerRpcReturnValue` was not possible?